### DISCUSSION TITLE
Webhooks integration with FastAPI

### DISCUSSION THREAD

**Discussion by hillairet** [SatelCreative/spylib#115] [Ideas]
How can we implement the FastAPI specific integration for webhooks?
And maybe decide for the rest of the library at the same time how we do these library specific integrations.

@qw-in Recommended to look at ariadne as an example.

**Comment by qw-in** [comment]
Ariadne uses two patterns:
- `from ariadne.asgi import GraphQL` for asgi support
- full separate library for Django at [`ariadne_django`](https://github.com/reset-button/ariadne_django)

Would be good to look at other examples too

**Comment by lishanl** [comment]
I like the first pattern better 
```py
from spylib.webhook import FastAPI
```
The folder structure mentioned here
https://github.com/SatelCreative/spylib/discussions/127#discussioncomment-2702307

**Reply by hillairet** [reply]
Maybe in that case to isolate we'd have:
```python
from spylib.webhook.fastapi import FastAPI
```
This way we can require that `fastapi` is installed when loading `spylib/webhook/fastapi.py` but not `spylib/webhook/__init__.py`
